embedded-pulsar/src/test/java/com/playtika/test/pulsar/EmbeddedPulsarBootstrapConfigurationTest.java

```diff
@@ -0,0 +1,34 @@
+package com.playtika.test.pulsar;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.BeanFactoryUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.testcontainers.containers.PulsarContainer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(classes = TestConfiguration.class,
+        properties = "embedded.pulsar.enabled=true")
+class EmbeddedPulsarBootstrapConfigurationTest {
```

Comment:
please, use single abstract test class for enabled pulsar, so only single spring contexts, hence single pulsar container would be started across all 3 tests